node/src/components/consensus/protocols/highway.rs

```diff
@@ -405,9 +406,24 @@ where
                 };
                 match pvv.timestamp() {
                     Some(timestamp) if timestamp > Timestamp::now() => {
-                        self.store_vertex_for_addition_later(timestamp, sender, pvv)
+                        // If it's not from an equivocator and from the future, add to queue
+                        if !pvv.is_signed_by_faulty() {
```

Comment:
Instead of adding new field to `PreValidatedVertex`, can we check here whether its creator is faulty?

Reply:
That's a good point! We're only using that information here, after all.
Also, the information can get outdated, so storing it in `PreValidatedVertex` isn't appropriate anyway.

Reply:
>Also, the information can get outdated

Didn't think about it but that's EXCELLENT point! ;)